domehunter/dome_control.py

```diff
@@ -0,0 +1,674 @@
+        # the rotation relay is the on/off switch for dome rotation
+        # the direction relay will toggle either the CW or CCW direction
+        # (using both the normally open and normally close relay terminals)
+        # so when moving the dome, first set the direction relay position
+        # then activate the rotation relay
+        self.rotation_relay = DigitalOutputDevice(
+            ROTATION_RELAY_PIN_NUMBER, initial_value=False)
+        self.direction_CW_relay = DigitalOutputDevice(
+            DIRECTION_RELAY_PIN_NUMBER, initial_value=False)
+        # because we initialiase the relay in the nnormally closed position
+        self.current_direction = "CCW"
+
+        # turn on the relay LEDs if we are debugging
+        if debug_lights:  # pragma: no cover
+            self._turn_led_on(leds=['relay_2_normally_closed'])
+            self._turn_led_on(leds=['relay_1_normally_closed'])
+
+###############################################################################
+# Properties
+###############################################################################
+
+    @property
+    def is_home(self):
+        """Send True if the dome is at home."""
+        return self._at_home
+
+    @property
+    def dome_in_motion(self):
+        """Send True if dome is in motion."""
+        # This property is set to false by the _stop_moving() private method
+        # and to set to true by the _move_cw() and _move_ccw() private methods
+        return self._dome_moving
+
+###############################################################################
+# Methods
+###############################################################################
+
+    """These map directly onto the AbstractMethods created by RPC."""
+
+    def abort(self):
+        """
+        Stop everything by switching the dome motor on/off relay to off.
+
+        """
+        # TODO: consider another way to do this in case the relay fails/sticks
+        # one way might be cut power to the automationHAT so the motor relays
+        # will receive no voltage even if the relay is in the open position?
+        self._stop_moving()
+
+    def getAz(self):
+        """
+        Return current Azimuth of the Dome.
+
+        Returns
+        -------
+        float
+            The Dome azimuth in degrees.
+
+        """
+        if self.dome_az is None:
+            print("Cannot return Azimuth as Dome is not yet calibrated.\
+                   Run calibration loop")
+        return self.dome_az
+
+    def GotoAz(self, az):
+        """
+        Send Dome to a requested Azimuth position.
+
+        Parameters
+        ----------
+        az : float
+            Desired dome azimuth position in degrees.
+
+        """
+        if self.dome_az is None:
+            print('Dome is not yet calibrated, running through calibration'
+                  ' procedure, then will go to AZ specified.')
+            self.calibrate_dome_encoder_counts()
```

Comment:
Yeah I suppose, could just have it trigger find_home() instead of having it go through the whole calibration routine. Then it would just proceed to go to the requested az.